packages/contracts/deploy/40_finalize-managing-dao/10_install-multisig-on-managing-dao.ts

```diff
@@ -0,0 +1,123 @@
+import {HardhatRuntimeEnvironment} from 'hardhat/types';
+import {DeployFunction} from 'hardhat-deploy/types';
+
+import buildMetadataJson from '../../src/plugins/governance/multisig/build-metadata.json';
+import {findEvent} from '../../utils/event';
+import {hashHelpers} from '../../test/test-utils/psp/hash-helpers';
+
+import {getContractAddress, managePermission} from '../helpers';
+
+interface Ehre extends HardhatRuntimeEnvironment {
+  aragonPluginRepos: {multisig: string};
+}
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const {getNamedAccounts, ethers} = hre;
+  const {deployer} = await getNamedAccounts();
+
+  const ehre = hre as Ehre;
+
+  // Get info from .env
+  const approvers =
+    process.env.MANAGINGDAO_MULTISIG_APPROVERS?.split(',') || [];
+  const minApprovals = process.env.MANAGINGDAO_MULTISIG_MINAPPROVALS || 0;
+  const listedOnly = process.env.MANAGINGDAO_MULTISIG_LISTEDONLY || false;
+
+  if (!approvers.length || !minApprovals || !listedOnly)
+    throw new Error(
+      `Some .env settings for managingDAO multisig are not set correctly, see:\n` +
+        `(MANAGINGDAO_MULTISIG_APPROVERS no. of addresses: ${approvers.length})\n` +
+        `(MANAGINGDAO_MULTISIG_MINAPPROVALS: ${minApprovals})\n` +
+        `(MANAGINGDAO_MULTISIG_LISTEDONLY: ${listedOnly})\n`
+    );
+
+  // Get `managingDAO` address.
+  const managingDAOAddress = await getContractAddress('DAO', hre);
+
+  // Get `DAO` contract.
+  const managingDaoContract = await ethers.getContractAt(
+    'DAO',
+    managingDAOAddress
+  );
+
+  // Get `PluginSetupProcessor` address.
+  const pspAddress = await getContractAddress('PluginSetupProcessor', hre);
+
+  // Get `PluginSetupProcessor` contract.
+  const pspContract = await ethers.getContractAt(
+    'PluginSetupProcessor',
+    pspAddress
+  );
+
+  // Installing multisig
+  const multisigRepoAddress = ehre.aragonPluginRepos.multisig;
+  const versionTag = [1, 1];
+  const pluginSetupRef = [versionTag, multisigRepoAddress];
+
+  // Prepare multisig plugin for managingDAO
+  const data = ethers.utils.defaultAbiCoder.encode(
+    buildMetadataJson.pluginSetupABI.prepareInstallation,
+    [approvers, [listedOnly, minApprovals]]
+  );
+  const prepareParams = [pluginSetupRef, data];
+  const prepareTx = await pspContract.prepareInstallation(
+    managingDAOAddress,
+    prepareParams
+  );
+  prepareTx.wait();
+
+  // extract info from prepare event
+  const event = await findEvent(prepareTx, 'InstallationPrepared');
+  const installationPreparedEvent = event.args;
+
+  // Grant `ROOT_PERMISSION` to `PluginSetupProcessor`.
+  await managePermission({
```

Comment:
use `applyMultiTargetPermissions` for 2 grant. then do the `apply` installation. then do `applyMultiTargetPermissions` again for 2 revoke.

Reply:
amended